pkg/network/ovn_kubernetes.go

```diff
@@ -1895,6 +2038,49 @@ func shouldUpdateOVNKonPrepull(ovn bootstrap.OVNBootstrapResult, releaseVersion
 	return true, false
 }
 
+// isUserIPsecMachineConfigPresent returns true if user owned MachineConfigs for IPsec plugin
+// already present in master and worker nodes, otherwise returns false.
+func isUserIPsecMachineConfigPresent(infra bootstrap.InfraStatus) bool {
+	if infra.MasterIPsecMachineConfig == nil && infra.WorkerIPsecMachineConfig == nil {
+		return false
+	}
+	return !containsNetworkOwnerRef(infra.MasterIPsecMachineConfig.OwnerReferences) ||
+		!containsNetworkOwnerRef(infra.WorkerIPsecMachineConfig.OwnerReferences)
+}
+
+func containsNetworkOwnerRef(ownerRefs []metav1.OwnerReference) bool {
+	for _, ownerRef := range ownerRefs {
+		if ownerRef.Kind == "Network" && (ownerRef.Controller != nil && *ownerRef.Controller) &&
+			ownerRef.Name == "cluster" {
+			return true
+		}
+	}
+	return false
+}
+
+// isIPsecMachineConfigActive checks if both master and worker's machine config pool are ready with
+// ipsec machine config extension rolled out then return true.
```

Comment:
```
// isIPsecMachineConfigActive returns true if both master and worker's machine config pool are ready with
// ipsec machine config extension rolled out, otherwise returns false.
```